src/types.ts

```diff
@@ -1,5 +1,10 @@
 export type Metadata = { [key: string]: string }
 
+export type MetaLike = {
```

Comment:
I think MetaLike should be named something more descriptive, it is not *like* meta. It contains meta!

What about `Meta` or `WithMeta` or `MetaInfo`?